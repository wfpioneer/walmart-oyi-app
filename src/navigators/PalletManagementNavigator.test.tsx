import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import ShallowRenderer from 'react-test-renderer/shallow';
import {
  PalletManagementNavigatorStack,
  renderManagePalletKebabButton,
  renderScanButton
} from './PalletManagementNavigator';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

jest.mock('../state/actions/Modal', () => ({
  showActivityModal: jest.fn(),
  hideActivityModal: jest.fn()
}));

jest.mock('../utils/AppCenterTool.ts', () => ({
  ...jest.requireActual('../utils/__mocks__/AppCenterTool'),
  trackEvent: jest.fn()
}));

describe('PalletManagement Navigator', () => {
  it('Renders the PalletManagement Navigator', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PalletManagementNavigatorStack
        isManualScanEnabled={false}
        managePalletMenu={false}
        dispatch={jest.fn()}
        createPallet={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the PalletManagement Navigator when create pallet is true', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PalletManagementNavigatorStack
        isManualScanEnabled={false}
        managePalletMenu={false}
        dispatch={jest.fn()}
        createPallet={true}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders and Calls the Manual Scan Button', () => {
    const mockDispatch = jest.fn();

    const { toJSON, getByTestId } = render(
      renderScanButton(mockDispatch, false)
    );
    const scanButton = getByTestId('barcode-scan');
    fireEvent.press(scanButton);

    expect(mockDispatch).toHaveBeenCalled();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Renders and Calls the Kebab Menu button', () => {
    const mockDispatch = jest.fn();

    const { toJSON, getByTestId } = render(
      renderManagePalletKebabButton(false, mockDispatch)
    );
    const palletMenu = getByTestId('pallet_menu');
    fireEvent.press(palletMenu);

    expect(mockDispatch).toHaveBeenCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
