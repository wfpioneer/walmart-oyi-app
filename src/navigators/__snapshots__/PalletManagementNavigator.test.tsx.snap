// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PalletManagement Navigator Renders and Calls the Kebab Menu button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="pallet_menu"
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingRight": 10,
      }
    }
  >
    <Image
      source=""
      style={
        {
          "height": 25,
          "marginRight": 10,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;

exports[`PalletManagement Navigator Renders and Calls the Manual Scan Button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="barcode-scan"
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 15,
      }
    }
  >
    <Icon
      color="#ffffffff"
      name="barcode-scan"
      size={20}
    />
  </View>
</View>
`;

exports[`PalletManagement Navigator Renders the PalletManagement Navigator 1`] = `
<StackNavigator
  screenOptions={
    {
      "headerMode": "float",
      "headerStyle": {
        "backgroundColor": "#0071e9",
      },
      "headerTintColor": "#ffffffff",
    }
  }
>
  <Screen
    component={[Function]}
    name="PalletManagementHome"
    options={
      {
        "headerRight": [Function],
        "headerTitle": [Function],
      }
    }
  />
  <Screen
    component={[Function]}
    name="ManagePallet"
    options={
      {
        "headerRight": [Function],
        "headerTitle": "[missing "en.PALLET.MANAGE_PALLET" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="CombinePallets"
    options={
      {
        "headerRight": [Function],
        "headerTitle": "[missing "en.PALLET.COMBINE_PALLETS" translation]",
      }
    }
  />
</StackNavigator>
`;

exports[`PalletManagement Navigator Renders the PalletManagement Navigator when create pallet is true 1`] = `
<StackNavigator
  screenOptions={
    {
      "headerMode": "float",
      "headerStyle": {
        "backgroundColor": "#0071e9",
      },
      "headerTintColor": "#ffffffff",
    }
  }
>
  <Screen
    component={[Function]}
    name="PalletManagementHome"
    options={
      {
        "headerRight": [Function],
        "headerTitle": [Function],
      }
    }
  />
  <Screen
    component={[Function]}
    name="ManagePallet"
    options={
      {
        "headerRight": [Function],
        "headerTitle": "[missing "en.PALLET.MANAGE_PALLET" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="CombinePallets"
    options={
      {
        "headerRight": [Function],
        "headerTitle": "[missing "en.PALLET.COMBINE_PALLETS" translation]",
      }
    }
  />
</StackNavigator>
`;
